### PR TITLE
[Merged by Bors] - refactor(Analysis/Complex): merge Complex/Convex and Convex/Complex

### PR DESCRIPTION
**Pull request by loefflerd**
This merges the two files `Analysis/Complex/Convex.lean` and `Analysis/Convex/Complex.lean`, putting lemmas about convexity in the complex plane together; and moves a lemma from `UpperHalfPlane/Topology.lean` to this file, substantially shortening the import path to all the C-star-algebras files.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Convex/Complex.lean [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-Authors: Yury Kudryashov, Yaël Dillies
--/
-import Mathlib.Analysis.Convex.Basic
-import Mathlib.Data.Complex.Module
-
-/-!
-# Convexity of half spaces in ℂ
-
-The open and closed half-spaces in ℂ given by an inequality on either the real or imaginary part
-are all convex over ℝ.
```
Comment (j-loreaux) [inline Mathlib/Analysis/Convex/Complex.lean:13]: Can you move this into the module doc of the other file?

**Review by j-loreaux — COMMENTED** [review]
Thanks! I appreciate this import reduction.

bors d+

**Comment by loefflerd** [comment]
bors d+

**Pull request closed without merging**